SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -349,6 +535,66 @@ void HGCDigitizer::accumulate(PileUpEventPrincipal const& e,
   }
 }
 
+//
+template <typename GEOM>
+void HGCDigitizer::accumulate_forPreMix(edm::Handle<edm::PCaloHitContainer> const& hits,
+                                        int bxCrossing,
+                                        const GEOM* geom,
+                                        CLHEP::HepRandomEngine* hre) {
+  if (nullptr == geom)
+    return;
+
+  float keV2fC(0.f);
+
+  int nchits = (int)hits->size();
+
+  std::vector<HGCCaloHitTuple_t> hitRefs;
+  hitRefs.reserve(nchits);
+  for (int i = 0; i < nchits; ++i) {
+    const auto& the_hit = hits->at(i);
+    DetId id = simToReco(geom, the_hit.id());
+    // to be written the verbosity block
+    if (id.rawId() != 0) {
+      hitRefs.emplace_back(i, id.rawId(), (float)the_hit.time());
+    }
+  }
+  std::sort(hitRefs.begin(), hitRefs.end(), this->orderByDetIdThenTime);
+
+  nchits = hitRefs.size();
+  for (int i = 0; i < nchits; ++i) {
+    const int hitidx = std::get<0>(hitRefs[i]);
+    const uint32_t id = std::get<1>(hitRefs[i]);
+    if (!validIds_.count(id))
+      continue;
+
+    HGCPUSimHitDataAccumulator::iterator simHitIt = pusimHitAccumulator_->emplace(id, HGCCellHitInfo()).first;
+    if (id == 0)
+      continue;
+
+    const float toa = std::get<2>(hitRefs[i]);
+    const PCaloHit& hit = hits->at(hitidx);
+    const float charge = hit.energy() * 1e6 * keV2fC;  // * getCCE(geom, id, cce_);
```

Comment:
it would be preferable to add a _keV operator in https://github.com/cms-sw/cmssw/blob/master/DataFormats/Math/interface/CMSUnits.h to do this conversion
@cvuosalo

Reply:
@kpedro88 Actually, this case calls for a conversion function like `convertGeVToKeV()` to be added to `CMSUnits.h.`, but energy literals would also be helpful.
I have created issue #29750 for these requests.